gradio/utils.py

```diff
@@ -612,3 +613,37 @@ def set_directory(path: Path):
 
 def strip_invalid_filename_characters(filename: str) -> str:
     return "".join([char for char in filename if char.isalnum() or char in "._- "])
+
+
+def sanitize_value_for_csv(value: str | Number) -> str | Number:
+    """
+    Sanitizes a value that is being written to a CSV file to prevent CSV injection attacks.
+    Reference: https://owasp.org/www-community/attacks/CSV_Injection
+    """
+    if isinstance(value, Number):
+        return value
+    unsafe_prefixes = ["=", "+", "-", "@", "\t", "\n"]
+    unsafe_sequences = [",=", ",+", ",-", ",@", ",\t", ",\n"]
+    if any(value.startswith(prefix) for prefix in unsafe_prefixes) or any(
+        sequence in value for sequence in unsafe_sequences
+    ):
+        value = "'" + value
+    return value
+
+
+def sanitize_list_for_csv(
+    values: List[str | Number] | List[List[str | Number]],
+) -> List[str | Number] | List[List[str | Number]]:
+    """
+    Sanitizes a list of values (or a list of list of values) that is being written to a
+    CSV file to prevent CSV injection attacks.
+    """
+    sanitized_values = []
+    for value in values:
+        if isinstance(value, list):
+            sanitized_value = sanitize_list_for_csv(value)
```

Comment:
I think a list comprehension would be better here than the recursion

```python
sanitized_value = [sanitize_value_for_csv(v) for v in value]
```

Reply:
Thanks, updated!